x-pack/legacy/plugins/maps/public/layers/styles/vector/components/vector_style_editor.js

```diff
@@ -64,32 +65,29 @@ export class VectorStyleEditor extends Component {
       };
     };
 
-    const dateFields = await this.props.layer.getDateFields();
-    const dateFieldPromises = dateFields.map(getFieldMeta);
-    const dateFieldsArray = await Promise.all(dateFieldPromises);
-    if (this._isMounted && !_.isEqual(dateFieldsArray, this.state.dateFields)) {
-      this.setState({ dateFields: dateFieldsArray });
+    //These are all fields (only used for text labeling)
+    const fields = await this.props.layer.getFields();
+    const fieldPromises = fields.map(getFieldMeta);
+    const fieldsArrayAll = await Promise.all(fieldPromises);
+    if (this._isMounted && !_.isEqual(fieldsArrayAll, this.state.fields)) {
```

Comment:
styleableFields, dateFieldsArray, and numberFieldsArray are all derived from fields. How about adding an `if` check to short circuit here instead of having this check for every single array?

```
async _loadFields() {
    const getFieldMeta = async field => {
      return {
        label: await field.getLabel(),
        name: field.getName(),
        origin: field.getOrigin(),
        type: await field.getDataType(),
      };
    };

    //These are all fields (only used for text labeling)
    const fields = await this.props.layer.getFields();
    const fieldPromises = fields.map(getFieldMeta);
    const fieldsArrayAll = await Promise.all(fieldPromises);

    if (!this._isMounted || _.isEqual(fieldsArrayAll, this.state.fields)) {
      return;
    }

    this.setState({
      fields: fieldsArrayAll,
      styleableFields: fieldsArrayAll.filter(field => {
        return CATEGORICAL_DATA_TYPES.includes(field.type) || ORDINAL_DATA_TYPES.includes(field.type);
      }),
      dateFields: fieldsArrayAll.filter(field => field.type === 'date'),
      numberFields: fieldsArrayAll.filter(field => field.type === 'number'),
    });
  }
```